packages/@react-aria/collections/src/BaseCollection.ts

```diff
@@ -65,8 +67,64 @@ export class CollectionNode<T> implements Node<T> {
     node.render = this.render;
     node.colSpan = this.colSpan;
     node.colIndex = this.colIndex;
+    node.filter = this.filter;
     return node;
   }
+
+  filter(collection: BaseCollection<T>, newCollection: BaseCollection<T>, filterFn: FilterFn<T>): CollectionNode<T> | null {
+    let [firstKey, lastKey] = filterChildren(collection, newCollection, this.firstChildKey, filterFn);
+    let newNode: Mutable<CollectionNode<T>> = this.clone();
+    newNode.firstChildKey = firstKey;
+    newNode.lastChildKey = lastKey;
+    return newNode;
+  }
+}
+
+// TODO: naming, but essentially these nodes shouldn't be affected by filtering (BaseNode)?
+// Perhaps this filter logic should be in CollectionNode instead and the current logic of CollectionNode's filter should move to Table
+export class FilterLessNode<T> extends CollectionNode<T> {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  filter(collection: BaseCollection<T>, newCollection: BaseCollection<T>, filterFn: FilterFn<T>): FilterLessNode<T> | null {
+    return this.clone();
+  }
+}
+
+export class ItemNode<T> extends CollectionNode<T> {
+  static readonly type = 'item';
+
+  constructor(key: Key) {
+    super(ItemNode.type, key);
+  }
```

Comment:
why static? I don't think we need this as a class field?
this should be enough if it's not overridable

```suggestion
  constructor(key: Key) {
    super('item', key);
  }
```

Reply:
I originally just had them like the above, but typescript was giving me all kinds of hell in CollectionBuilder as a result due to the overrides

Reply:
hmmm... because it only knew 'type = item'?
or what were the errors?